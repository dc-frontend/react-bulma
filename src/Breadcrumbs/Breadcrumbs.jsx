import React from 'react'
import PropTypes from 'prop-types'

/**
 * Breadcrumb (Bulma)
 * @param {*} props
 */
export function Breadcrumb (props) {
  if (props.active) {
    return <li className='is-active' key={`${props.name}-breadcrumb`}><a href='#' aria-current='page'>{props.name}</a></li>
  }
  return <li key={`${props.name}-breadcrumb`}><a href={props.href}>{props.name}</a></li>
}

Breadcrumb.propTypes = {
  active: PropTypes.bool,
  name: PropTypes.string.isRequired,
  href: PropTypes.string
}

/**
 * Breadcrumbs (Bulma)
 * @param {*} props
 */
export function Breadcrumbs (props) {
  return <nav className='breadcrumb has-succeeds-separator' aria-label='breadcrumbs'>
    <ul>{props.children}</ul>
  </nav>
}

Breadcrumbs.propTypes = {
  children: PropTypes.any
}
Breadcrumbs.defaultProps = {
  class: 'has-succeeds-seperator'
}
