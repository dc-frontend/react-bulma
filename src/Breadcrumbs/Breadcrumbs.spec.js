// Link.react.test.js
import React from 'react'
import { Breadcrumb } from './Breadcrumbs'
import renderer from 'react-test-renderer'

test('Breadcrumb renders as expected', () => {
  const component = renderer.create(
    <Breadcrumb href='/' name='Home'></Breadcrumb>
  )

  expect(component.toJSON()).toMatchSnapshot()
})
