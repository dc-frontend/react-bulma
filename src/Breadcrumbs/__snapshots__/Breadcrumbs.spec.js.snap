// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb renders as expected 1`] = `
<li>
  <a
    href="/"
  >
    Home
  </a>
</li>
`;
